gusto/diagnostics.py

```diff
@@ -401,6 +401,42 @@ def name(self):
         return self.field1+"_perturbation"
 
 
+class Precipitation(DiagnosticField):
+    name = "Precipitation"
+
+    def setup(self, state):
+        if not self._initialised:
+            space = state.spaces("DG1", state.mesh, "DG", 1)
+            super(Precipitation, self).setup(state, space=space)
+
+            rain = state.fields('rain')
+            rho = state.fields('rho')
+            v = state.fields('rainfall_velocity')
+            phi = TestFunction(space)
+            flux = TrialFunction(space)
+            n = FacetNormal(state.mesh)
+            un = 0.5 * (dot(v, n) + abs(dot(v, n)))
+            self.flux = Function(space)
+            self.total_water = Function(space)
+
+            if space.extruded:
+                self.ds = ds_b + ds_t + ds_v
+            else:
+                self.ds = ds
+
+            a = phi * flux * dx
+            L = phi * rain * un * rho * self.ds
```

Comment:
So this is diagnosing the flux into the top and out of the bottom?

Reply:
Yes, although I am expecting there only to be flux out and not in!